### PR TITLE
(feat) Add ability to pass schema to wr.s3.read_parquet

### PR DESCRIPTION
**Pull request by kukushking**
### Relates
- #1662 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awswrangler/_arrow.py [hunk 1]
```diff
@@ -76,6 +76,9 @@ def _table_to_df(
     if table.schema.metadata is not None and b"pandas" in table.schema.metadata:
         metadata = json.loads(table.schema.metadata[b"pandas"])
 
+    if "schema" in kwargs:
+        kwargs.pop("schema")
```
Comment (kukushking) [inline awswrangler/_arrow.py:80]: There's a bit of a mess between different pyarrow calls regarding which arguments are supported. I'm opening this draft PR to show a couple of examples, like `schema` or `coerce_int96_timestamp_unit` that we handle as a separate arg

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 4ad3088a63c9524bc00258c69ebd8bbd7ef876c3
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=633fd274-d958-45e2-9e66-9a37f6e04d22%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-s6u9F3qN9oFy
* Commit ID: 4ad3088a63c9524bc00258c69ebd8bbd7ef876c3
* Result: SUCCEEDED
* [Build Logs](https://pciyom2pbh.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f1c3109b-7b61-41a4-9b1d-3e28252a664e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: c7baf9c0a6c70720facfe81207f3737ec30163af
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f2f45cc9-046b-4e57-b8e7-d0a2a4f71465%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by kukushking** [comment]
Closing this for now due to ray limiting us to pyarrow <=7.

**Pull request closed without merging**